src/shims/unix/fd.rs

```diff
@@ -360,13 +414,14 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
         let fd = this.read_scalar(fd_op)?.to_i32()?;
 
-        let Some(file_descriptor) = this.machine.fds.remove(fd) else {
-            return Ok(Scalar::from_i32(this.fd_not_found()?));
-        };
-        let result = file_descriptor.close(this.machine.communicate())?;
-        // return `0` if close is successful
-        let result = result.map(|()| 0i32);
-        Ok(Scalar::from_i32(this.try_unwrap_io_result(result)?))
+        Ok(Scalar::from_i32(if let Some(file_descriptor) = this.machine.fds.remove(fd) {
```

Comment:
Please bring back the `let...else`, that's much easier to read.
Why did you even remove that?